### PR TITLE
Fix resize animations in Dialog

### PR DESCRIPTION
**Pull request by MrFlashAccount**
### Pull Request Description

This PR fixes the Resize animations in Dialog component:

1. Removes resize for initial mount / fullscreen dialogs
2. Fixes measuring the content size
3. Fixes bugs in `useMeasure` hook
4. Adds memoization for Text and Loader components (because of react-compiler and because this components accept only primitive values)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.
- [ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

File: app/gui/src/dashboard/App.tsx [hunk 1]
```diff
@@ -532,42 +520,40 @@ function AppRouter(props: AppRouterProps) {
 
   return (
     <FeatureFlagsProvider>
-      <MotionConfig reducedMotion="user" transition={DEFAULT_TRANSITION_OPTIONS}>
```
Comment (MrFlashAccount) [inline app/gui/src/dashboard/App.tsx:535]: Here we only move the MotionConfig from `<App />` to `<UIProvider />` to reuse it in `Storybook`

File: app/gui/src/dashboard/hooks/measureHooks.ts [hunk 2]
```diff
@@ -74,21 +75,29 @@ export interface Options {
...
 
+  const [ref, forceRefresh] = useMeasureCallback({ ...options, onResize: onResizeStableCallback })
+
+  return [ref, bounds, forceRefresh] as const
+}
+
+/**
+ * Helper hook that uses motion primitive to optimize renders, works best with motion components
+ */
+export function useMeasureSignal(options: Options = {}) {
```
Comment (MrFlashAccount) [inline app/gui/src/dashboard/hooks/measureHooks.ts:90]: Doesn't work yet with `layout` prop, see: https://github.com/motiondivision/motion/issues/2907

File: app/gui/src/dashboard/index.tsx [hunk 3]
```diff
@@ -116,23 +116,21 @@ export function run(props: DashboardProps) {
     reactDOM.createRoot(root).render(
       <React.StrictMode>
         <QueryClientProvider client={queryClient}>
-          <DevtoolsProvider>
```
Comment (MrFlashAccount) [inline app/gui/src/dashboard/index.tsx:119]: Devtools provider no longer needed due to using the `zustand`

File: app/gui/src/dashboard/components/AriaComponents/Dialog/Modal.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
+/**
+ *
+ */
+export interface ModalProps {}
+
+/**
+ *
+ */
+export function Modal() {
```
Comment (somebody1234) [inline app/gui/src/dashboard/components/AriaComponents/Dialog/Modal.tsx]: what is this component for?

**Review by somebody1234 — APPROVED** [review]
CR :white_check_mark:

File: app/gui/src/dashboard/components/AriaComponents/Dialog/Modal.tsx (hunk 4 above)
Comment (MrFlashAccount) [reply app/gui/src/dashboard/components/AriaComponents/Dialog/Modal.tsx]: It's removed, wrong abstraction

**Pull request merged**